src/scopes/activityReport/singleOrMultiRecipient.js

```diff
@@ -0,0 +1,30 @@
+import { Op } from 'sequelize';
+import { sequelize } from '../../models';
+
+const recipientSql = (singleOrMulti) => (`SELECT
+  "ActivityRecipients"."activityReportId"
+  FROM "Recipients" "Recipients"
+  INNER JOIN "Grants" "Grants"
+  ON "Grants"."recipientId" = "Recipients"."id"
+  INNER JOIN "ActivityRecipients" "ActivityRecipients"
+    ON "ActivityRecipients"."grantId" = "Grants"."id"
+  GROUP BY "ActivityRecipients"."activityReportId"
+  HAVING
+    COUNT(DISTINCT CASE WHEN "Recipients"."uei" IS NULL OR  "Recipients"."uei" = ''
+      THEN
+        "Recipients"."name"
+      ELSE "Recipients"."uei"
+      END)
```

Comment:
This would be about 30% faster
```suggestion
    COUNT(DISTINCT COALESCE(NULLIF("Recipients"."uei",''),"Recipients"."name"))
```

Reply:
Ok this was my initial implementation but I couldn't get it to work guess I was missing the NULLIF.

Good to know.

Thanks.